lib/Dancer2/Core/Request.pm

```diff
@@ -653,6 +653,14 @@ sub _shallow_clone {
     return $new_request;
 }
 
+
+sub route {
+    my ($self, $route) = @_;
+    if ($route) { $self->{route_matched} = $route; }
+
+    return $self->{route_matched};
```

Comment:
I'm not sure this is the best way to accomplish this. This creates a hashref slot without an appropriate attribute and thus introduces state we cannot account for. It also makes `route` a `rw`, which we are trying to avoid doing.

Why can't `route` be a `ro` attribute and that's it?